pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,93 @@
+package dns
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		record *iov1.DNSRecord
+		zones  []configv1.DNSZone
+		expect operatorv1.ConditionStatus
+	}{
+		{
+			name: "testing for a successful update of 1 record",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: operatorv1.ConditionFalse,
+		},
+		{
+			name: "when no zones available",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones:  []configv1.DNSZone{{}},
+			expect: operatorv1.ConditionTrue,
+		},
+		{
+			name:   "when no region available",
+			record: nil,
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: operatorv1.ConditionTrue,
+		},
+	}
+	for _, test := range tests {
+		r := &reconciler{
+			dnsProvider: &FakeDNSProvider{},
+		}
+		t.Logf("evaluating test %s", test.name)
+		actual, results := r.publishRecordToZones(test.zones, test.record)
+		_ = results
+		for _, dnsStatus := range actual {
+			for _, condition := range dnsStatus.Conditions {
```

Comment:
Make sure the right number of conditions is returned.  For this unit test, I think it would be acceptable to assume that the number of conditions will equal the number of zones.

Reply:
Sorry I did not get it but now I have tested multiple zones as well and the test cases are working fine. But happy to learn from any feedback given.

Reply:
No need to compare lengths if you use `cmp.Equal` as I describe in a comment below.